05_casts/05_00_Interesting_cases/05_00_00_up_and_down_casts.cpp

```diff
@@ -0,0 +1,44 @@
+// Show how static_cast works, and also that there is no check in it,
+// so the last case is possible.
+
+#include <iostream>
+
+struct base {
+    void foo() {
+        std::cout << "base foo" << std::endl;
+    }
+};
+
+struct direved : base {
+    void foo() {
+        std::cout << "direved foo" << std::endl;
+    }
+
+    void bar() {
+        std::cout << "direved bar" << std::endl;
+    }
+};
+
+int main() {
+    base* base_ptr = new base();
+
+    base_ptr->foo();
+    delete base_ptr;
+
+    base_ptr = new direved();
```

Comment:
Either description should cover why the example is so long or it should be reduced.